### DISCUSSION TITLE
Why does not set `QUARTO_JULIA_PROJECT=.` by default in quarto project?

### DISCUSSION THREAD

**Discussion by eitsupi** [quarto-dev/quarto-cli#9986] [Q&A] [answered]
### Description

I'm trying out the new Julia engine, and I should add a `_environment` file for julia engine to recognize the Julia project in the `qmd` file when `_quarto.yml` and `Project.toml` are in the root directory and the `qmd` file is in a subdirectory.
Given that the Julia REPL starts in the root directory on VS Code, I find it compelling that `QUARTO_JULIA_PROJECT=.` is set by default, but is there any reason not to do this?

**Comment by mcanouil** [comment]
@cscheid @jkrumbiegel @MichaelHatherly

**Reply by mcanouil** [reply, marked as answer]
FYI, there is an ongoing PR to improve the documentation:
- https://github.com/quarto-dev/quarto-web/pull/1102

**Reply by MichaelHatherly** [reply]
Yeah, if there's anything that isn't clearly answered in that PR change then I'm sure @jkrumbiegel can expand the changes there.

**Reply by eitsupi** [reply]
Oh, thanks.
As it says in that document, it actually seems to work fine without any settings because the runner automatically detects the Julia Project (I guess I was trying different things and missed it)

**Comment by dmi3kno** [comment]
I came here to say that no other settings, other than
```yml
julia:
  exeflags: ["--project=."]
```
make sense. I don't understand what environment does Quarto use by default. Is there a way that Quarto could be explicit which environment it discovered by using `--project=@` heuristic?

**Reply by mcanouil** [reply]
>  I don't understand what environment does Quarto use by default, but it is a mess!

@dmi3kno You can disagree with some design choices but please be mindful and respectful.
This kind of comment does not help and is not constructive.

FYI, the native Julia engine was brought by the community with the help of Quarto team.

> [!IMPORTANT]
> We appreciate your interest in the Quarto project, but the content above violates our [Code of Conduct](https://github.com/quarto-dev/quarto-cli?tab=coc-ov-file#readme).
>
> As a member of the Quarto community, you are expected to follow our Code of Conduct, which outlines the standards of behavior and communication that we value. The Code of Conduct applies to all online interactions within the Quarto project, and it is enforced by the Quarto team. By participating in the Quarto community, you agree to abide by the Code of Conduct and respect the rights and dignity of others. You can find the Code of Conduct at <https://github.com/quarto-dev/quarto-cli?tab=coc-ov-file#readme>.
> 
> Thank you for your cooperation and contribution. 🙏

**Reply by dmi3kno** [reply]
i take it back. Sorry! My bad

**Reply by cderv** [reply]
Just to clarify some context here: 

- Quarto julia engine support is done through QuartoNotebookRunner.jl project : https://github.com/PumasAI/QuartoNotebookRunner.jl
   This dependency is included in Quarto as a specific version to install in a private environment for Quarto. From time to time, the version 	 
   is updated. Our documentation also document how one could use a greater version of this dependency. 
   
- As with other engine, this dependency is responsible for running any Julia computation, and send it back to Quarto for processing. So Quarto does not really set a default `--project` - QuartoNotebookRunner.jl is doing that. 

- They recently change their default: https://github.com/PumasAI/QuartoNotebookRunner.jl/issues/664 and `--project=@.` is the default (and not `--project=@` which is different). 

This is quite advanced knowledge how those tools integrated, but I understand how it can be important to understand for anyone developing some Julia tool to be used with Quarto too. We'll try to improve our advanced doc in soon future. 

Hope this helps understand.
